src/lib/sidenav/sidenav.ts

```diff
@@ -110,7 +112,22 @@ export class MdSidenav implements AfterContentInit {
    * @param _elementRef The DOM element reference. Used for transition and width calculation.
    *     If not available we do not hook on transitions.
    */
-  constructor(private _elementRef: ElementRef) {}
+  constructor(private _elementRef: ElementRef) {
+    this.onOpen.subscribe(() => {
+      this._elementFocusedBeforeSidenavWasOpened = document.activeElement as HTMLElement;
```

Comment:
I don't think Angular has a way of getting this, at the moment. There are other components that use `document.activeElement` as well. 

AFAIK when we start supporting Universal, we can check if the renderer is a browser via:

```
import { isBrowser, isNode } from 'angular2-universal'
```